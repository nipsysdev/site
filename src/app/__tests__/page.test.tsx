import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import RootPage from '../page';

// Mock the LoadSequence component
vi.mock('@/components/LoadSequence', () => ({
  default: () => <div data-testid="load-sequence">LoadSequence Component</div>,
}));

describe('RootPage', () => {
  it('should render the LoadSequence component', () => {
    render(<RootPage />);

    expect(screen.getByTestId('load-sequence')).toBeInTheDocument();
    expect(screen.getByText('LoadSequence Component')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<RootPage />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
