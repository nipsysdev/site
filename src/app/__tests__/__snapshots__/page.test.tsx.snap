// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RootPage > should match snapshot 1`] = `
<div
  data-testid="load-sequence"
>
  LoadSequence Component
</div>
`;
